src/Workspaces/Core/Portable/Shared/Extensions/TextSpanExtensions.cs

```diff
@@ -61,5 +63,21 @@ public static bool IsAround(this TextSpan span, SyntaxNodeOrToken startNode, Syn
             var outerSpan = TextSpan.FromBounds(startNode.FullSpan.Start, endNode.FullSpan.End);
             return span.Contains(innerSpan) && outerSpan.Contains(span);
         }
+
+        public static IEnumerable<TextSpan> Subtract(this TextSpan span, TextSpan except)
+        {
+            var startSegmentEnd = Math.Min(span.End, except.Start);
+            if (span.Start < startSegmentEnd)
+                yield return TextSpan.FromBounds(span.Start, startSegmentEnd);
+
+            var endSegmentStart = Math.Max(span.Start, except.End);
+            if (endSegmentStart < span.End)
+                yield return TextSpan.FromBounds(endSegmentStart, span.End);
+        }
```

Comment:
consider writing explicit unit tests for this.

Reply:
Forgot, will add to TODO. Think this is buggy.